Database/CleanArchive.py

```diff
@@ -88,12 +96,69 @@ def clean(all=True):
         transaction = {
             '$pull': {'snippets': {'transcript': ''}}
         }
+        if duplicate < 1.0:
+            print('Duplicate found')
+            set_fields['duplicate'] = True
+            duplicates['metadata.Subtitle'] += 1
+        else:
+            set_fields['duplicate'] = False
         if errors:
             set_fields['errors'] = errors
         else:
             transaction['$unset'] = {'errors': 1}
         if set_fields:  # $set cannot be empty.
             transaction['$set'] = set_fields
+
+
+
         db.ArchiveIndex.update_one({'_id': episode['_id']}, transaction)
     print('Updates:', updates)
     print('Failures:', failures)
+    print('Duplicates', duplicates)
+
+
+def checkDuplicate(epi):
+    upperBound = db.ArchiveIndex.find({'identifier': epi['identifier']})[0]['date']
+    dateTimeObj = datetime.datetime.strptime(upperBound, '%Y-%m-%dT%H:%M:%SZ')
+    lowerBound = str(dateTimeObj - dateTimeObj - datetime.timedelta(days=1))
```

Comment:
identifier is the same as _id but _id has an index (hash map) for faster queries.
Use find_one to get only the first result (or none if there are no results). The search will exit early after finding just 1 result, speeding up the query.
Use the metadata.Datetime_UTC field. It's stored as a datetime already and the date field given by archive (the one you're using) can be off by up to a day.
Also maybe increase the timedelta to 2-4 days.
Why is lowerBound a str(x-x-1)?

Reply:
I don't see that field in metadata.

Reply:
I see it is being added by the cleaning function but when comparing the episodes that field hasnt been set yet

Reply:
It shouldn't be added by the cleaning function. It's added in the build archive step. _id is the mongoDB version of primary key. If none is specified, it generates a random key object.

Reply:
I don't see it in this example

{"Topics": ["Washington", "Biden", "Opec", "Iran", "China", "New York", "Harlem", "St. Petersburg Florida", "Mexico", "South Korea", "Obama Administration", "Scott Brown", "Matt Miller", "Hagel", "Yang Yang", "Crumpton", "Ford", "Wti", "Nebraska Panhandle", "New Data"], "Network": "Bloomberg", "Duration": "01:01:00", "Scanned in": "San Francisco, CA, USA", "Source": "Comcast Cable", "Tuner": "Channel v106", "Title": "Bloomberg", "Subtitle": "Bloomberg December 4, 2013 7:00pm-8:01pm EST", "Date": "December 5, 2013"}

Reply:
It's not in the metadata. It's a field in the root.